test/token-assert.js

```diff
@@ -370,6 +386,252 @@ describe('modules/token-assert', function() {
                 nextToken: undefined
             });
         });
+
+        describe('exact', function() {
+            it('should trigger error on too few newlines', function() {
+                var file = createJsFile('x\n=y;');
+
+                var tokenAssert = new TokenAssert(file);
+                var onError = sinon.spy();
+                tokenAssert.on('error', onError);
+
+                var tokens = file.getTokens();
+                tokenAssert.differentLine({
+                    token: tokens[0],
+                    nextToken: tokens[1],
+                    exactly: 2
+                });
+
+                assert(onError.calledOnce);
+
+                var error = onError.getCall(0).args[0];
+                assert.equal(error.message, 'x and = should be exactly 2 line(s) apart');
+            });
+
+            it('should trigger error on too many specified newlines', function() {
+                var file = createJsFile('x\n\n\n\n=y;');
+
+                var tokenAssert = new TokenAssert(file);
+                var onError = sinon.spy();
+                tokenAssert.on('error', onError);
+
+                var tokens = file.getTokens();
+                tokenAssert.differentLine({
+                    token: tokens[0],
+                    nextToken: tokens[1],
+                    exactly: 2
+                });
+
+                assert(onError.calledOnce);
+                var error = onError.getCall(0).args[0];
+                assert.equal(error.message, 'x and = should be exactly 2 line(s) apart');
+            });
+
+            it('should not trigger error on correct specified newlines', function() {
+                var file = createJsFile('x\n=y;');
+
+                var tokenAssert = new TokenAssert(file);
+                var onError = sinon.spy();
+                tokenAssert.on('error', onError);
+
+                var tokens = file.getTokens();
+                tokenAssert.differentLine({
+                    token: tokens[0],
+                    nextToken: tokens[1],
+                    exactly: 1
+                });
+
+                assert(!onError.calledOnce);
+            });
+
+            it('should not trigger error on multiple specified newlines negative', function() {
+                var file = createJsFile('x\n=y;');
+
+                var tokenAssert = new TokenAssert(file);
+                var onError = sinon.spy();
+                tokenAssert.on('error', onError);
+
+                var tokens = file.getTokens();
+                tokenAssert.differentLine({
+                    token: tokens[1],
+                    nextToken: tokens[0],
+                    exactly: 1
+                });
+
+                assert(!onError.calledOnce);
+            });
```

Comment:
`!onError.called` here and everywhere you're using once to test that it wasn't called.